src/components/Header.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useEffect } from "react";
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  Button,
+  IconButton,
+  MenuItem,
+  Drawer,
+} from "@mui/material";
+import MenuIcon from "@mui/icons-material/Menu";
+import { makeStyles } from "@mui/styles";
+import { Link as RouterLink, Outlet } from "react-router-dom";
+
+const useStyles = makeStyles(() => ({
+  header: {
+    backgroundColor: "#400CCC",
+    paddingRight: "79px",
+    paddingLeft: "118px",
+    "@media (max-width: 900px)": {
+      paddingLeft: 0,
+    },
+  },
+  logo: {
+    fontFamily: "Work Sans, sans-serif",
+    fontWeight: 600,
+    color: "#FFFEFE",
+    textAlign: "left",
+  },
+  menuButton: {
+    fontFamily: "Open Sans, sans-serif",
+    fontWeight: 700,
+    size: "18px",
+    marginLeft: "38px",
+  },
+  toolbar: {
+    display: "flex",
+    justifyContent: "space-between",
+  },
+  drawerContainer: {
+    padding: "20px 30px",
+  },
+}));
+
+const headerData = [
+  {
+    label: "Portfolio",
+    href: "/",
+  },
+  {
+    label: "Add Stock",
+    href: "/add",
+  },
+];
+
+const Header = () => {
+  const [viewState, setViewState] = useState({
+    mobileView: false,
+    drawerOpen: false,
+  });
+
+  const { mobileView, drawerOpen } = viewState;
+  const { header, logo, menuButton, toolbar, drawerContainer } = useStyles();
+
+  useEffect(() => {
+    const setResponsiveness = () => {
+      return window.innerWidth < 900
+        ? setViewState((prevState) => ({ ...prevState, mobileView: true }))
+        : setViewState((prevState) => ({ ...prevState, mobileView: false }));
+    };
+
+    setResponsiveness();
+    window.addEventListener("resize", () => setResponsiveness());
+
+    return () => {
+      window.removeEventListener("resize", () => setResponsiveness());
+    };
+  }, []);
+
+  const displayDesktop = () => {
+    return (
+      <Toolbar className={toolbar}>
+        {trackMyShitLogo()}
+        <div>{getMenuButtons()}</div>
+      </Toolbar>
+    );
+  };
+
+  const getDrawerChoices = () => {
+    return headerData.map(({ label, href }) => {
+      return (
+        <RouterLink
+          {...{
+            component: RouterLink,
+            to: href,
+            color: "inherit",
+            style: { textDecoration: "none" },
+            key: label,
+          }}
+        >
+          <MenuItem>{label}</MenuItem>
+        </RouterLink>
+      );
+    });
+  };
+
+  const displayMobile = () => {
+    const handleDrawerOpen = () =>
+      setViewState((prevState) => ({ ...prevState, drawerOpen: true }));
+    const handleDrawerClose = () =>
+      setViewState((prevState) => ({ ...prevState, drawerOpen: false }));
+    return (
+      <Toolbar>
+        <IconButton
+          {...{
```

Comment:
do we need the spread on these props

Reply:
Oops went full copy pasta there. I'll fix it tomorrow.